cmd/run.go

```diff
@@ -87,6 +100,16 @@ func runRun(cmd *cobra.Command, args []string, clientFn runClientFn) (err error)
 
 	client := clientFn(config)
 
+	if !function.Built() {
+		build, err := cmd.Flags().GetBool("build")
+		if err == nil && build {
+			if err := client.Build(cmd.Context(), config.Path); err != nil {
+				return err
+			}
+		} else if err != nil {
+			fmt.Printf("Function will not be built: %v\n", err)
+		}
+	}
```

Comment:
```suggestion
	if config.Build && !function.Built() {
		if err = client.Build(cmd.Context(), config.Path); err != nil {
			return
		}
	}
```

Reply:
Since some of these commands have quite a few options, we have been bundling them up into a "config" object, and doing all the err checks, nil checks, and validation there.  That way, when it's time to actually use the values, we can just refer to `config.X`.  This seems to have cleaned things up a lot, especially in these more complex commands.

Therefore, it might be nice to add a `.Build` boolean member to the RunConfig object, populated when `newRunConfig(cmd)` is called on line 76